resources/views/components/comment/list.blade.php

```diff
@@ -0,0 +1,128 @@
+if ($user) {
+    $subjectType = SubscriptionSubjectType::fromArticleType($articleType);
+    if ($subjectType !== null) {
+        $canSubscribe = true;
+        $isSubscribed = isUserSubscribedToArticleComments($articleType, $articleId, $user->id);
+    }
+}
+
+$route = match((int) $articleType) {
+    ArticleType::Game => route('game.comments', ['game' => $articleId]),
+    ArticleType::GameModification => route('game.modification-comments', ['game' => $articleId]),
+    ArticleType::GameHash => route('game.hashes.comments', ['game' => $articleId]),
+    ArticleType::SetClaim => route('game.claims.comments', ['game' => $articleId]),
+    ArticleType::Achievement => route('achievement.comments', ['achievement' => $articleId]),
+    ArticleType::Leaderboard => route('leaderboard.comments', ['leaderboard' => $articleId]),
+    ArticleType::User => route('user.comments', ['user' => $article ?? User::find($articleId)]),
+    ArticleType::UserModeration => route('user.moderation-comments', ['user' => $article ?? User::find($articleId)]),
+    default => 'unsupported type ' . $articleType,
+};
+
+@endphp
+
+<div class="commentscomponent">
+    <div class="flex justify-between items-center mb-3">
+        <div>
+            @if ($totalComments == 0)
+                <i>No comments</i>
+            @elseif (!$embedded)
+                @if ($totalPages > 1)
+                    <div class="w-full flex items-center">
+                        <x-paginator :totalPages="$totalPages" :currentPage="$currentPage" />
+                    </div>
+                @endif
+            @elseif ($totalComments > count($comments))
+                <div class="hidden sm:block">
+                    Recent comments: <span class="smalltext">(<a href="{{ $route }}">All {{ $totalComments }}</a>)</span>
+                </div>
+
+                <div class="sm:hidden">
+                    <p>Recent comments</p>
+                    <p class="smalltext">(<a href="{{ $route }}">See all {{ $totalComments }}</a>)</p>
+                </div>
+            @else
+                Comments:
+            @endif
+        </div>
+
+        @if ($canSubscribe)
+            <x-update-subscription-button
+                :subjectType="$subjectType"
+                :subjectId="$articleId"
+                :isSubscribed="$isSubscribed"
+                resource="{{ $embedded ? 'comments' : null }}"
+            />
+        @endif
+    </div>
+
+    <table id="feed" class="table-highlight">
+        <tbody>
+            @foreach ($comments as $comment)
+                <x-comment.item
+                    :author="$comment->user"
+                    :when="$comment->Submitted"
+                    :payload="nl2br($comment->Payload)"
+                    :articleType="$articleType"
+                    :articleId="$articleId"
+                    :commentId="$comment->ID"
+                    :allowDelete="$user->can('delete', [App\Models\Comment::class, $comment])"
```

Comment:
If `$user` is null, the page will crash. Try visiting /game/1 whlie unauthenticated.

Reply:
good catch.